include/openthread/srp_client.h

```diff
@@ -0,0 +1,501 @@
+ * interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns The lease interval (in seconds).
+ *
+ */
+uint32_t otSrpClientGetLeaseInterval(otInstance *aInstance);
+
+/**
+ * This function sets the lease interval used in SRP update requests.
+ *
+ * Changing the lease interval does not impact the accepted lease interval of already registered services/host-info.
+ * It only affects any future SRP update messages (i.e., adding new services and/or refreshes of the existing services).
+ *
+ * @param[in] aInstance   A pointer to the OpenThread instance.
+ * @param[in] aInterval   The lease interval (in seconds).
+ *
+ */
+void otSrpClientSetLeaseInterval(otInstance *aInstance, uint32_t aInterval);
+
+/**
+ * This function gets the key lease interval used in SRP update requests.
+ *
+ * Note that this is the lease duration requested by the SRP client. The server may choose to accept a different lease
+ * interval.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns The key lease interval (in seconds).
+ *
+ */
+uint32_t otSrpClientGetKeyLeaseInterval(otInstance *aInstance);
+
+/**
+ * This function sets the key lease interval used in SRP update requests.
+ *
+ * Changing the lease interval does not impact the accepted lease interval of already registered services/host-info.
+ * It only affects any future SRP update messages (i.e., adding new services and/or refreshes of existing services).
+ *
+ * @param[in] aInstance    A pointer to the OpenThread instance.
+ * @param[in] aInterval    The key lease interval (in seconds).
+ *
+ */
+void otSrpClientSetKeyLeaseInterval(otInstance *aInstance, uint32_t aInterval);
+
+/**
+ * This function gets the host info.
+ *
+ * @param[in] aInstance        A pointer to the OpenThread instance.
+ *
+ * @returns A pointer to host info structure.
+ *
+ */
+const otSrpClientHostInfo *otSrpClientGetHostInfo(otInstance *aInstance);
+
+/**
+ * This function sets the host name label.
+ *
+ * After a successful call to this function, `otSrpClientCallback` will be called to report the status of host info
+ * registration with SRP server.
+ *
+ * The name string buffer pointed to by @p aName MUST persist and stay unchanged after returning from this function.
+ * OpenThread will keep the pointer to the string.
+ *
+ * The host name can be set before client is started or after start but before host info is registered with server
+ * (host info should be in either `STATE_TO_ADD` or `STATE_REMOVED`).
+ *
+ * @param[in] aInstance   A pointer to the OpenThread instance.
+ * @param[in] aName       A pointer to host name label string (MUST NOT be NULL). Pointer to the string buffer MUST
+ *                        persist and remain valid and constant after return from this function.
+ *
+ * @retval OT_ERROR_NONE            The host name label was set successfully.
+ * @retval OT_ERROR_INVALID_ARGS    The @p aName is NULL.
```

Comment:
We internationally don't perform such checks in client code in the APIs. I  skipped and/or removed many such checks in the client implementation to reduce its code size.

Note that bad input like this will still be detected, for example the name label  length is checked during SRP update message preparation and/or bad chars in name are checked by server (I expect) and then rejected. So the user will be notified in the callback.

Reply:
Agree that it's a good practice for clients to ignore excess checking and depend on server to do more comprehensive validations.